### PR TITLE
refactor: modified Args type

### PR DESCRIPTION
**Pull request by ssi02014**
## Summary
I modified the `Args` type, which was defined in several places, to be uniform. 

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: src/vanilla/atom.ts [hunk 1]
```diff
@@ -5,9 +5,7 @@ type Setter = <Value, Args extends unknown[], Result>(
   ...args: Args
 ) => Result
 
-type SetAtom<Args extends unknown[], Result> = <A extends Args>(
-  ...args: A
-) => Result
```
Comment (ssi02014) [inline src/vanilla/atom.ts:10]: After checking the usage of `SetAtom`, it seems that the `<A extends Args>` part is unnecessary.

File: src/react/useAtom.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import type {
 import { useAtomValue } from './useAtomValue.ts'
 import { useSetAtom } from './useSetAtom.ts'
 
-type SetAtom<Args extends any[], Result> = (...args: Args) => Result
+type SetAtom<Args extends unknown[], Result> = (...args: Args) => Result
```
Comment (ssi02014) [inline src/react/useAtom.ts:11]: I thought the `unknwon` was more stable than the `any` 
https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type

Comment (ssi02014) [reply src/react/useAtom.ts:11]: Also, `Args extends unknown[]` is already used in many places. 🤗

**Review by dai-shi — APPROVED** [review]
AFAIR, there were some cases that we need to use `any`, but if tests are passing, it should be okay. (Or, we will see issues reported later..)

**Pull request merged**